v2/outbound_transport_middleware.go

```diff
@@ -72,17 +72,28 @@ func applyUnaryOutboundTransportMiddleware(o UnaryOutbound, f UnaryOutboundTrans
 	return unaryOutboundWithMiddleware{o: o, f: f}
 }
 
-// UnaryOutboundTransportMiddlewareFunc adapts a function into a UnaryOutboundTransportMiddleware.
-type UnaryOutboundTransportMiddlewareFunc func(context.Context, *Request, *Buffer, UnaryOutbound) (*Response, *Buffer, error)
+// UnaryOutboundTransportMiddlewareStruct adapts a function and name into a
+// UnaryOutboundTransportMiddleware.
+type UnaryOutboundTransportMiddlewareStruct struct {
+	MiddlewareName string
```

Comment:
optional: `Middleware` might be redundant / is inconsistent with the other named types.

Reply:
yeah, unfortunately `Name` can't  be a field name and a method name